python/lsst/daf/butler/registries/sql.py

```diff
@@ -415,9 +415,21 @@ def registerDatasetType(self, datasetType: DatasetType) -> bool:
         _, inserted = self._managers.datasets.register(datasetType)
         return inserted
 
-    def removeDatasetType(self, name: str) -> None:
+    def removeDatasetType(self, name: str | tuple([str, ...])) -> None:
         # Docstring inherited from lsst.daf.butler.registry.Registry
-        self._managers.datasets.remove(name)
+
+        for datasetTypeExpression in ensure_iterable(name):
+            datasetTypes = list(self.queryDatasetTypes(datasetTypeExpression))
+            if not datasetTypes:
+                _LOG.info(f"Dataset type {datasetTypeExpression} not defined")
```

Comment:
The general comment is that all python logging should use deferred variable expansion. Should be something like:

```python
_LOG.info("Dataset type %r not defined", datasetTypeExpression)
```
where I use `%r` so that the string will be quoted in the log message.